src/zoom_func.h

```diff
@@ -68,6 +68,26 @@ static inline int UnScaleGUI(int value)
 	return UnScaleByZoom(value, ZOOM_LVL_GUI);
 }
 
+/**
+ * Scale zoom level relative to GUI zoom.
+ * @param value zoom level to scale
+ * @return scaled zoom level
+ */
+static inline ZoomLevel ScaleZoomGUI(ZoomLevel value)
+{
+	return ZoomLevel(value + (ZOOM_LVL_GUI - ZOOM_LVL_OUT_4X));
```

Comment:
I guess this method is similar to ScaleGUITrad in that it scales constants defined for the original/traditional zoom level.
Semantically ZOOM_LVL_OUT_4X is an arbitrary value, which changes meaning, if we ever add more zoom levels. The traditional zoom level is ZOOM_LVL_BASE.
```suggestion
	return ZoomLevel(value + (ZOOM_LVL_GUI - ZOOM_LVL_BASE));
```
Maybe the `ScaleZoomGUI` could be named in a ways, so it groups/associated better to the existing ScaleGUITrad, since both are used in similar context.

Reply:
I originally picked `ZOOM_LVL_OUT_4X` because that's the constant that's being defaulted to by the video driver:

https://github.com/OpenTTD/OpenTTD/blob/2d2a5dd9661f842b0399d367883e6bbf1f19cdca/src/video/video_driver.hpp#L176-L178

However I'm fine with the change; maybe we should update the video driver's default as well then?

Reply:
> Maybe the ScaleZoomGUI could be named in a ways, so it groups/associated better to the existing ScaleGUITrad, since both are used in similar context.

How about `ScaleZoomGUITrad` and `UnScaleZoomGUITrad`?

Reply:
> I guess this method is similar to ScaleGUITrad in that it scales constants defined for the original/traditional zoom level.

Indeed, however the default viewport's `ZoomLevel`s are defined assuming a GUI zoom of `ZOOM_LVL_OUT_4X`. 

> Semantically ZOOM_LVL_OUT_4X is an arbitrary value, which changes meaning, if we ever add more zoom levels. The traditional zoom level is ZOOM_LVL_BASE.

`ZOOM_LVL_BASE` is currently defined to be `2` while I need `ZOOM_LVL_OUT_4X` (currently defined as `4`), as that's what's being used here:

https://github.com/OpenTTD/OpenTTD/blob/2d2a5dd9661f842b0399d367883e6bbf1f19cdca/src/video/video_driver.hpp#L178

Reply:
Yeah, video_driver.hpp would also break if there would be more zoom levels added.

Reply:
Resolving this as I'm currently using the same constant as the method I'm matching.